### PR TITLE
Raw CA correction buffersize fixes

### LINKED ISSUE #16007: Darktable 4.6: Raw Chromatic Aberration causes munmap_chuck(): invalid pointer crash with SRW files and unhandled installtation exception
### Describe the bug

Version 4.0 was so far working without problems. Now I have updated darktable 4.0 => 4.6. 
I got the information that an unhandled exception occured (backtrace is uploaded).
However, starting darktable and working with lighttable is running well. 
But **darktable is shutting down by selecting the darkroom view for a raw picture**. 
In between I got once the notice that there is an OpenCL initilization issue but OpenCL support is not activated (screenshot preferences > processing is uploaded). For a jpg pcture I have so far no shutdown issue.

### Steps to reproduce

1. start darktable
2. click on raw picture to develop in the darkroom view
3. darktable is shutting down.

### Expected behavior

darktable should allow picture development in darkroom view

### Logfile | Screenshot | Screencast

[darktable_bt_NTW5G2.txt](https://github.com/darktable-org/darktable/files/13812080/darktable_bt_NTW5G2.txt)
darktable-pref_processing

### Commit

_No response_

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.6.0 install file darktable-4.6.0-win64.exe, downloaded on Dec 28, 2023

### What OS are you using?

Windows

### What is the version of your OS?

Windows 10 Home 22H2 Build 19045.3803

### Describe your system?

Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz   2.71 GHz 16GB
On-Board Graphic Intel® HD Graphics 530

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Reproducible with RAW pictures, not with Jpeg pictures.
The steps above reproducible with a fresh edit.
Lua scripts are not in use.

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
Due to rounding we need slightly larger internal buffers in some cases, reporting issue was 

Fixes #16007.

### CODE REVIEW

File: src/iop/cacorrect.c [hunk 1]
```diff
@@ -281,7 +281,15 @@ void process(
   float *RawDataTmp = NULL;
   float *Gtmp = NULL;
 
-  float *out = dt_alloc_align_float(roi_in->width * roi_in->height);
+  const size_t width = roi_in->width;
+  const size_t height = roi_in->height;
+  const size_t ibsize = dt_round_size(width, DT_CACHELINE_BYTES) * (height + 2);
```
Comment (ralfbrown) [inline src/iop/cacorrect.c:286]: DT_CACHELINE_FLOATS would actually be sufficient here, but it's not that big an increase in memory allocation to use \_BYTES

Comment (jenshannoschwalm) [reply src/iop/cacorrect.c]: Right :-) fixed

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by TurboGit** [comment]
For this to merge in 4.6.x branch I had to also import into the branch:

```
commit 3cfe5c926c2348dec745042c94000563a8c025f8
Author: Hanno Schwalm <hanno@schwalm-bremen.de>
Date:   Sun Dec 24 07:07:05 2023 +0100

    deduplicate processed_maximum usage
```

Seems pretty safe, let me know if this decision is wrong.

**Comment by jenshannoschwalm** [comment]
Safe for me too :-)

**Comment by jenshannoschwalm** [comment]
Release note for 4.6 and 4.8

Fix crashes in raw CA correction due to improper internal memory allocation

**Pull request merged**